packages/ra-core/src/form/FormGroupContextProvider.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { ReactNode, useEffect } from 'react';
+import { FormGroupContext } from './FormGroupContext';
+import { useFormContext } from './useFormContext';
+
+/**
+ * This provider allows its input children to know they belong to a specific group.
+ * This enables other components to access groups properties such as its
+ * validation (valid/invalid) or whether its inputs have been updated (dirty/pristine).
+ * @param props The component props
+ * @param {ReactNode} props.children The form group content
+ * @param {String} props.name The form group name
+ */
+export const FormGroupContextProvider = ({
+    children,
+    name,
+}: {
+    children: ReactNode;
+    name: string;
+}) => {
+    const formContext = useFormContext();
+
```

Comment:
You should check that the formContext contains the `register` and `unregister` functions and throw a meaningful error if it doesn't. 

I'm surprised that TypeScript didn't warn you on this one.